src/main/scala/dsptools/numbers/chisel_types/DspRealTypeClass.scala

```diff
@@ -47,6 +47,7 @@ trait DspRealSigned extends Any with Signed[DspReal] with hasContext {
     ComparisonHelper(a === DspReal(0.0), a < DspReal(0.0))
   }
   def abs(a: DspReal): DspReal = Mux(a < DspReal(0.0), DspReal(0.0) - a, a)
+  def context_abs(a: DspReal): DspReal = Mux(a < DspReal(0.0), DspReal(0.0) - a, a)
```

Comment:
a < DspReal(0.0), a need to be delayed by context.numAddPipes (since DspReal(0.0) - 1 uses that number of add registers. Delayed via Shift. Otherwise there will be horrible timing mismatch.